src/main/java/com/google/error_analyzer/backend/Database.java

```diff
@@ -0,0 +1,160 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.*/
+
+package com.google.error_analyzer.backend;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.http.HttpHost;
+import java.io.IOException;
+import java.util.*;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.index.query.MatchQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.query.SimpleQueryStringBuilder;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.search.fetch.subphase.highlight.HighlightBuilder;
+import org.elasticsearch.search.fetch.subphase.highlight.HighlightBuilder.Field;
+import org.elasticsearch.search.fetch.subphase.highlight.HighlightField;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.admin.indices.get.GetIndexRequest;
+import org.elasticsearch.action.admin.indices.get.GetIndexRequest;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.query.QueryBuilder;
+
+public class Database implements DaoInterface {
+
+    private static final RestHighLevelClient client = 
+    new RestHighLevelClient(RestClient.builder(new HttpHost("localhost", 9200, "http")));
+    private static final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+    private static final int windowSize = 10;
+    private static final Logger logger = LogManager.getLogger(Database.class);
+
+    //search db using keywords and return searchHits having highlight field added 
+    public ArrayList < SearchHit > fullTextSearch(String fileName, String searchString, String field) throws IOException {
+        int offset = 0;
+        SearchHit[] searchHits = null;
+        ArrayList < SearchHit > searchResult = new ArrayList();
+        while (searchHits == null || searchHits.length != 0) {
+            SearchRequest searchRequest = new SearchRequest(fileName);
+            SimpleQueryStringBuilder simpleQueryBuilder = QueryBuilders.simpleQueryStringQuery(searchString);
+            searchSourceBuilder.query(simpleQueryBuilder).size(windowSize).from(offset);
```

Comment:
Just a thought: Maybe it is better to keep the limit on the result size rather than windowSize. It will speed up the search.
This comment is not actionable.